test_util/marathon.py

```diff
@@ -205,6 +205,85 @@ def _pool_for_marathon_app(app_id):
             raise Exception("Application deployment failed - operation was not "
                             "completed in {} seconds.".format(timeout))
 
+    def deploy_pod(self, pod_definition):
+        """Deploy a pod to marathon
+
+        This function deploys an a pod and then waits for marathon to
+        aknowledge it's successfull creation or fails the test.
+
+        It waits until all the instances reach tasksRunning and then tasksHealthy state.
+
+        Args:
+            pod_definition: a dict with pod definition as specified in
+                            Marathon API
+            check_health: wait until Marathon reports tasks as healthy before
+                          returning
+        Returns:
+            Scaling instance count
+        """
+        timeout = 120
+
+        r = self.post('v2/pods', json=pod_definition)
+        assert r.ok, 'status_code: {} body: {}'.format(r.status_code, r.body)
+        logging.info('Response from marathon: {}'.format(repr(r.json())))
+
+        @retrying.retry(wait_fixed=2000, stop_max_delay=timeout * 1000,
+                        retry_on_result=lambda ret: ret is False,
+                        retry_on_exception=lambda x: False)
+        def _wait_for_pod_deployment(pod_id):
+            # TODO(greggomann): Revisit this logic. Can we make a request for
+            # info on only our specific deployment? See DCOS-11707.
+            r = self.get('v2/deployments')
+            data = r.json()
+            if len(data) > 0:
+                logging.info('Waiting for pod to be deployed %s', repr(data))
```

Comment:
@greggomann please send a follow-up PR which fixes this, landing as it is currently.